### PR TITLE
pybind: support mon target in pybind

### PR DESCRIPTION
**Pull request by Liuchang0812**
[The first commit](https://github.com/ceph/ceph/pull/15409/commits/7b5aaf53262ab3921c33849c478be9106f38293f) supports mon target in pybind.

I fix some format problems based on `PEP8` in [the second commit](https://github.com/ceph/ceph/pull/15409/commits/d0c4d92832f61aaf2af62e4f8469486a430673ba). I will revert the second commit if it's inappropriate.

@tchaikov mind taking a look?

### CODE REVIEW

**Comment by Liuchang0812** [comment]
local test result:
```
➜  build git:(wip-support-mon-target-in-pybind) ✗ ctest -R "cephtool-test-*"   --output-on-failure
Test project /home/liuchang/WorkSpace/ceph/build
    Start 4: cephtool-test-mds.sh
1/4 Test #4: cephtool-test-mds.sh .............   Passed   68.90 sec
    Start 5: cephtool-test-mon.sh
2/4 Test #5: cephtool-test-mon.sh .............   Passed  1459.36 sec
    Start 6: cephtool-test-osd.sh
3/4 Test #6: cephtool-test-osd.sh .............   Passed   24.87 sec
    Start 7: cephtool-test-rados.sh
4/4 Test #7: cephtool-test-rados.sh ...........   Passed  196.65 sec

100% tests passed, 0 tests failed out of 4

Total Test time (real) = 1749.83 sec
```

File: src/ceph.in [hunk 1]
```diff
@@ -618,24 +627,23 @@ def complete(sigdict, args, target):
...
 def ping_monitor(cluster_handle, name, timeout):
+    """Ping a monitor
+    """
     if 'mon.' not in name:
         print('"ping" expects a monitor to ping; try "ping mon.<id>"', file=sys.stderr)
         return 1
 
     mon_id = name[len('mon.'):]
-    if (mon_id == '*') :
+    if (mon_id == '*'):
```
Comment (tchaikov) [inline src/ceph.in]: since you at this, could you remove the parentheses?

File: src/ceph.in [hunk 2]
```diff
@@ -618,24 +627,23 @@ def complete(sigdict, args, target):
     return 0
 
 
-###
-# ping a monitor
-###
 def ping_monitor(cluster_handle, name, timeout):
+    """Ping a monitor
```
Comment (tchaikov) [inline src/ceph.in]: could drop this comment, it is barely helpful.

File: src/ceph.in [hunk 3]
```diff
@@ -890,14 +896,16 @@ def main():
             timeout = 5
 
         hdr('Monitor commands:')
-        print('[Contacting monitor, timeout after %d seconds]' % timeout)
+        if verbose:
+            print('[Contacting monitor, timeout after %d seconds]' % timeout)
+        return do_extended_help(parser, childargs, ('mon', ''), ' '.join(childargs))
```
Comment (tchaikov) [inline src/ceph.in]: is this part of the pep8 change?

File: src/ceph.in [hunk 4]
```diff
@@ -913,17 +921,14 @@ def main():
         print(str(e), file=sys.stderr)
         return 1
 
-    if parsed_args.help:
```
Comment (tchaikov) [inline src/ceph.in]: ditto.

**Comment by Liuchang0812** [comment]
jenkins test this please

File: src/ceph.in (hunk 4 above)
Comment (Liuchang0812) [inline src/ceph.in]: it is a bug. i removed this change. it should be split to another PR.

```
[root@node44 /data/seal/deploy]# ceph ping --help

 Monitor commands: 
 =================
[Contacting monitor, timeout after 5 seconds]
"ping" requires a monitor name as argument: "ping mon.<id>"
```

File: src/ceph.in (hunk 3 above)
Comment (Liuchang0812) [inline src/ceph.in]: oh, sorry. i think it's an improvement.  ceph always prints `[Contacting monitor, timeout after 5 seconds]` when user typed `ceph --help`, as following:

```
[root@node44 /data/seal/deploy]# ceph osd --help
[a lot of help messages]

 Monitor commands:
 =================
[Contacting monitor, timeout after 5 seconds]
2017-06-02 11:55:28.437591 7f49598db700 -1 WARNING: the following dangerous and experimental features are enabled: bluestore,rocksdb
auth add <entity> {<caps> [<caps>...]}   add auth info for <entity> from input
                                          file, or random key if no input is
                                          given, and/or any caps specified in
                                          the command
```

File: src/ceph.in (hunk 1 above)
Comment (Liuchang0812) [inline src/ceph.in]: addressed

File: src/ceph.in (hunk 2 above)
Comment (Liuchang0812) [inline src/ceph.in]: addressed

File: src/ceph.in (hunk 4 above)
Comment (tchaikov) [inline src/ceph.in]: @Liuchang0812 ping?

Comment (Liuchang0812) [inline src/ceph.in]: yes, we print `Monitor commands: ===== [Contac..........]` here, without any command explain.  the problem is that:

1. we print header of help command at first
2. we check `ping` command. ( `ceph ping --help` exits here)
2. we print boby of help command.

we should put  1 and 3 together.

Comment (tchaikov) [inline src/ceph.in]: @Liuchang0812 sorry for the confusion, i mean, we should not include this change in the pep8 commit.

Comment (Liuchang0812) [inline src/ceph.in]: : ) I will remove it soon

Comment (Liuchang0812) [inline src/ceph.in]: done.

**Comment by Liuchang0812** [comment]
```
➜  build git:(wip-support-mon-target-in-pybind) ✗ ctest -R "cephtool-test-*"   --output-on-failure
Test project /home/liuchang/WorkSpace/ceph/build
    Start 4: cephtool-test-mds.sh
1/4 Test #4: cephtool-test-mds.sh .............   Passed   78.04 sec
    Start 5: cephtool-test-mon.sh
2/4 Test #5: cephtool-test-mon.sh .............   Passed  1449.85 sec
    Start 6: cephtool-test-osd.sh
3/4 Test #6: cephtool-test-osd.sh .............   Passed   18.97 sec
    Start 7: cephtool-test-rados.sh
4/4 Test #7: cephtool-test-rados.sh ...........   Passed  199.65 sec

100% tests passed, 0 tests failed out of 4

Total Test time (real) = 1746.57 sec
➜  build git:(wip-support-mon-target-in-pybind) ✗ pep8 ../src/ceph.in --max-line-length=180       
../src/ceph.in:118:1: E402 module level import not at top of file
../src/ceph.in:119:1: E402 module level import not at top of file
../src/ceph.in:120:1: E402 module level import not at top of file
../src/ceph.in:121:1: E402 module level import not at top of file
../src/ceph.in:122:1: E402 module level import not at top of file
../src/ceph.in:123:1: E402 module level import not at top of file
../src/ceph.in:124:1: E402 module level import not at top of file
../src/ceph.in:125:1: E402 module level import not at top of file
../src/ceph.in:127:1: E402 module level import not at top of file
../src/ceph.in:132:1: E402 module level import not at top of file
```

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-05_14:47:27-rados-wip-sage-testing-distro-basic-smithi/
